pkg/agent/downloader.go

```diff
@@ -77,14 +80,41 @@ func (d *Downloader) download(modelName string, storageUri string) error {
 	return nil
 }
 
+func (d* Downloader) GetProvider() (string, error) {
+	storageUri := ""
+	matches, _ := filepath.Glob(d.ModelDir + "/*.yaml")
```

Comment:
not sure if I understand this, the storage uri should be passed on `ModelSpec` and does not need to be parsed from the yaml.

Reply:
My understanding was that we'd configure the storage provider before any of the models are pulled, however I think I was over complicating things. We can just add each supported provider to the Provider map and then upon an add event, the correct provider will be parsed from the `ModelSpec` like you said. I'll go ahead and remove this logic.